lib/datadog/tracing/contrib/sinatra/tracer_middleware.rb

```diff
@@ -18,15 +18,9 @@ def initialize(app, opt = {})
             @app_instance = opt[:app_instance]
           end
 
-          # rubocop:disable Metrics/AbcSize
           # rubocop:disable Metrics/MethodLength
           def call(env)
-            # Set the trace context (e.g. distributed tracing)
-            if configuration[:distributed_tracing] && Tracing.active_trace.nil?
-              original_trace = Tracing::Propagation::HTTP.extract(env)
-              Tracing.continue_trace!(original_trace)
-            end
```

Comment:
I got these [`build_and_test_integration` errors](https://app.circleci.com/pipelines/github/DataDog/dd-trace-rb/13253/workflows/8a67b3dd-a5be-44b6-a9e9-d2d758d58e35/jobs/496520) and was puzzled for a bit.

The error was simple, that I forgot to update the line `original_trace = Tracing::Propagation::HTTP.extract(env)` with the move of `HTTP`, but I examined this error closer and more importantly why none of our tests exercised this code path, except for `build_and_test_integration`.

I believe this block, which runs when Sinatra is the top-level span, is not needed anymore, given we've introduced a `sinatra/framework.rb` file that always instruments Rack when Sinatra integration is activated: https://github.com/DataDog/dd-trace-rb/blob/916472f9db3d73fa92ad068df7495f25c7a45f86/lib/datadog/tracing/contrib/sinatra/framework.rb#L14-L28